generator/internal/genclient/language/internal/rust/rust.go

```diff
@@ -272,13 +272,25 @@ func (c *Codec) FieldAttributes(f *genclient.Field, state *genclient.APIState) [
 				slog.Error("missing key or value in map field")
 				return attributes
 			}
-			keyFormat := c.fieldFormatter(key)
-			valFormat := c.fieldFormatter(value)
+			keyFormat := c.fieldFormatter(key.Typez)
+			valFormat := c.fieldFormatter(value.Typez)
 			if keyFormat == "_" && valFormat == "_" {
 				return attributes
 			}
 			return append(attributes, fmt.Sprintf(`#[serde_as(as = "std::collections::HashMap<%s, %s>")]`, keyFormat, valFormat))
 		}
+		if f.TypezID == ".google.protobuf.BytesValue" {
```

Comment:
I would move this to a separate function and use a switch statement:

```
var formatter string
switch f.TypezID {
    case ".google.protobuf.BytesValue":
		formatter := c.fieldFormatter(genclient.BYTES_TYPE)
                 return []string{fmt.Sprintf(`#[serde_as(as = "Option<%s>")]`, formatter)}
	case ".google.protobuf.UInt64Value":
		formatter := c.fieldFormatter(genclient.UINT64_TYPE)
                 return []string{fmt.Sprintf(`#[serde_as(as = "Option<%s>")]`, formatter)}
	case f f.TypezID == ".google.protobuf.Int64Value":
		formatter := c.fieldFormatter(genclient.INT64_TYPE)
                 return []string{fmt.Sprintf(`#[serde_as(as = "Option<%s>")]`, formatter)}
}
```

Reply:
Good idea, thanks. Done.